tests/unit/lms/services/h_api_test.py

```diff
@@ -221,26 +86,8 @@ def test_if_given_custom_headers__api_request_adds_them(self, h_api, requests):
             {"X-Header": sentinel.header}
         )
 
-    def test__api_request_raises_HAPINotFoundError_for_404(self, h_api, requests):
-        response = Response()
-        response.status_code = 404
-        exception = RequestException(response=response)
-        requests.request.return_value.raise_for_status.side_effect = exception
-
-        with pytest.raises(HAPINotFoundError) as exc_info:
-            h_api._api_request(sentinel.method, "dummy-path")
-
-        # It records the requests exception that caused the HAPIError.
-        assert exc_info.value.__cause__ == exception
-
-    @pytest.mark.parametrize(
-        "exception_class",
-        [requests_.ConnectionError, HTTPError, ReadTimeout, TooManyRedirects],
-    )
```

Comment:
No point in enumerating different sub-classes of RequestException. We catch them all